### PR TITLE
update sample readme for the new structure

### PR DESCRIPTION
**Pull request by gaoning777**
Also temporarily disabled side_car sample due to its flaky behavior in sample test.

This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2058)

### CODE REVIEW

File: samples/README.md [hunk 1]
```diff
@@ -1,27 +1,62 @@
...
 * Build and deploy your pipeline [using the provided samples](https://www.kubeflow.org/docs/guides/pipelines/pipelines-samples/).
 
-
-
-
-This page tells you how to use the _basic_ sample pipelines contained in the repo.
+# Sample Structure
+The samples are organized into the core set and the contrib set. 
+
+**Core samples** demonstrates the full KFP functionalities and covered by the sample test infra.
```
Comment (numerology) [inline samples/README.md]: nit: are covered

File: samples/README.md [hunk 2]
```diff
@@ -1,27 +1,62 @@
...
+platform support that is hard to support in our test infra.
+
+In the Core directory, each sample will be in a separate directory. 
+In the Contrib directory, there is an intermediate directory for each contributor, 
+e.g. ibm and arena, within which each sample is in a separate directory. 
+An example of the resulting structure is as follows: 
+```
+pipelines/samples/
+Core/
+	DSL Static Type Checking/
```
Comment (numerology) [inline samples/README.md]: shall we adopt a more pythonic naming convention here? like 'dsl_static_type_checking'?
I'm a bit of worrying that adding spaces in dir name might be error-prone.

File: samples/README.md (hunk 1 above)
Comment (gaoning777) [reply samples/README.md]: Updated.

File: samples/README.md (hunk 2 above)
Comment (gaoning777) [reply samples/README.md]: Updated.

Comment (gaoning777) [reply samples/README.md]: I copied these from the PR description: https://github.com/kubeflow/pipelines/pull/1710

**Pull request merged**